docs/Actions.md

```diff
@@ -214,26 +217,184 @@ class ApproveButton extends Component {
+-   showNotification: PropTypes.func,
 };
 
-export default connect(null, {
-    showNotification,
-+   fetchStart,
-+   fetchEnd,
-    push,
-})(ApproveButton);
+-export default connect(null, {
+-    showNotification,
+-    push,
+-})(ApproveButton);
++export default withDataProvider(ApproveButton)
 ```
 
-That solution is perfectly all right from a UI perspective, but also a bit verbose. Fortunately, react-admin uses and provides a shorter way to make HTTP requests in a component.
+## Handling Side Effects
 
-## Using a Custom Action Creator
+Fetching data is called a *side effect*, since it calls the outside world, and is asynchronous. Usual actions may have other side effects, like showing a notification, or redirecting the user to another page. The `dataProvider` function injected by `withDataProvider` accepts a fourth parameter, which lets you describe the options of the query, including success and failure side effects. So the previous component can be even further simplified as follows:
 
-React-admin components don't call the `dataProvider` directly. Instead, they dispatch a Redux action with the `fetch` meta. React-admin watches this kind of actions, turns them into `dataProvider` calls, and handles the loading state automatically. You can use the same feature for your own actions.
+```diff
+// in src/comments/ApproveButton.js
+import {
+-  showNotification,
+   UPDATE,
+   withDataProvider,
+} from 'react-admin';
+-import { push } from 'react-router-redux';
+
+class ApproveButton extends Component {
+    handleClick = () => {
+        const { dataProvider, dispatch, record } = this.props;
+        const updatedRecord = { ...record, is_approved: true };
+-       dataProvider(UPDATE, 'comments', { id: record.id, data: updatedRecord })
+-           .then(() => {
+-               dispatch(showNotification('Comment approved'));
+-               dispatch(push('/comments'));
+-           })
+-           .catch((e) => {
+-               dispatch(showNotification('Error: comment not approved', 'warning'))
+-           });
+-   }
++       dataProvider(UPDATE, 'comments', { id: record.id, data: updatedRecord }, {
++           onSuccess: {
++               notification: 'Comment approved',
++               redirectTo: '/comments',
++           },
++           onError: {
++               notification: { body: 'Error: comment not approved', level: 'warning' }
++           }
++       })
+
+    render() {
+        return <FlatButton label="Approve" onClick={this.handleClick} />;
+    }
+}
+
+ApproveButton.propTypes = {
+    dataProvider: PropTypes.func.isRequired,
+-   dispatch: PropTypes.func.isRequired,
+    record: PropTypes.object,
+};
+
+export default withDataProvider(ApproveButton);
+```
+
+React-admin can handle the following side effects:
+
+- `notification`: Display a notification. The property value should be an object describing the notification to display. The `body` can be a translation key. `level` can be either `info` or `warning`.
```

Comment:
Probably worth mentionning the message args too

Reply:
A bit too long to describe for here. We'll have to write that down in another section, and place a reference here. To be done in a future PR.